colossalai/nn/optimizer/distributed_adafactor.py

```diff
@@ -0,0 +1,259 @@
+import math
+import os 
+from typing import Dict
+
+import torch
+from torch.optim import Optimizer
+import torch.distributed as dist
+
+from colossalai.shardformer.layer._operation import _gather
+from colossalai.tensor.d_tensor import (
+    is_distributed_tensor,
+    get_layout,
+    get_sharding_spec
+)
+
+
+# DistributedAdaFactor (with Tensor parallel and Zero stage 2)
+__all__ = ["DistributedAdaFactor"]
+class DistributedAdaFactor(Optimizer):
+    def __init__(
+        self,
+        params,
+        lr=None,
+        eps=(1e-30, 1e-3),
+        clip_threshold=1.0,
+        decay_rate=-0.8,
+        beta1=None,
+        weight_decay=0.0,
+        scale_parameter=True,
+        relative_step=True,
+        warmup_init=False,
+    ):
+        if lr is not None and relative_step:
+            raise ValueError("Cannot combine manual `lr` and `relative_step=True` options")
+        if warmup_init and not relative_step:
+            raise ValueError("`warmup_init=True` requires `relative_step=True`")
+
+        defaults = {
+            "lr": lr,
+            "eps": eps,
+            "clip_threshold": clip_threshold,
+            "decay_rate": decay_rate,
+            "beta1": beta1,
+            "weight_decay": weight_decay,
+            "scale_parameter": scale_parameter,
+            "relative_step": relative_step,
+            "warmup_init": warmup_init,
+        }
+        super().__init__(params, defaults)
+        self.tensor_parallel_size = 1
+        self.tensor_parallel_group = None
+        self.data_parallel_size = 1
+        self.data_parallel_group = 1
+        self.shard_to_param = None # Dict{id:shape}, sample {id(param): torch.tensor}
+        
+        
+    def setup_distributed(self, 
+                         tensor_parallel_group: dist.ProcessGroup = None, 
+                         data_parallel_group: dist.ProcessGroup = None,
+                         shard_to_param: Dict = None
+                         )-> None:
+        """
+        Inject features to the Optimizer
+        Args:
+            tensor_parallel_group (dist.ProcessGroup): The devices group for tensor parallel
+            data_parallel_group (dist.ProcessGroup): The devices group for data parallel
+            sharding_spec_dict (Dict{id(param):ShardingSpec}): ShardingSpecs of Each params
+            param_shape  (Dict{id(param):shape}): Paramater Shape of Each params
```

Comment:
Wrong docstr

Reply:
Fixed in https://github.com/hpcaitech/ColossalAI/pull/5484/commits/3194d11ed3d5423a2c7ed5a03a898f16a481c2c8;
Remove the description of class to make it look normal;